lib/mayaUsd/resources/ae/usd-shared-components/src/python/usdSharedComponents/collection/includeExcludeWidget.py

```diff
@@ -87,4 +107,73 @@ def setIncludeAll(self, value: bool):
         self._include.cbIncludeAll.setChecked(value)
 
     def onIncludeAllToggle(self):
-        self._collection.GetIncludeRootAttr().Set(self._include.cbIncludeAll.isChecked())
\ No newline at end of file
+        self._collection.GetIncludeRootAttr().Set(self._include.cbIncludeAll.isChecked())
+
+class EventFilter(QObject):
+    def __init__(self, widget, control):
+        super().__init__(widget)
+        self._widget = widget
+        self.control = control
+        self.widget.installEventFilter(self)
+        self.widget.setAcceptDrops(True)
+
+    @property
+    def widget(self):
+        return self._widget
+
+    def eventFilter(self, obj: QObject, event: QEvent):
+        if event.type() == QEvent.Drop:
+            mime_data = event.mimeData()
+            self.addItemToCollection(mime_data.text())
+            return True
+        elif event.type() == QEvent.DragEnter:
+            event.acceptProposedAction()
+            return True
+        elif event.type() == QEvent.DragMove:
+            event.acceptProposedAction()
+            return True
+
+        return super().eventFilter(obj, event)
+
+    def addItemToCollection(self, items):
+        itemList = items.split('\n')
+        for item in itemList:
+            path = ""
+            if ',' in item:
+                path = item.split(',')[1]
+            else:
+                path = item
+            
+            if not self._validatePath(self.control._collection, path):
+                return
+
+            if self.control._collection is not None:
+                if self.widget.headerTitle == "Include":
+                    self.control._collection.GetIncludesRel().AddTarget(path)
+                elif self.widget.headerTitle == "Exclude":
+                    self.control._collection.GetExcludesRel().AddTarget(path)
+
+        self.control.update()
+    '''
+    def removeItemToCollection(self, item):
+        if self.control._collection is not None:
+            if self.widget.headerTitle == "Include":
+                self.control._collection.GetIncludesRel().RemoveTarget(item)
+            elif self.widget.headerTitle == "Exclude":
+                self.control._collection.GetExcludesRel().RemoveTarget(item)
+
+        self.control.update()
+    '''
+
+    def _validatePath(self, collection, path):
+       
+        if not Sdf.Path.IsValidPathString(path):
+            raise ValueError("Invalid sdf path: " + path)
+
+        stage = self.control._collection.GetPrim().GetStage()
+        prim = stage.GetPrimAtPath(Sdf.Path(path))
+
+        if not prim or not prim.IsValid():
+            raise ValueError("Value must be a float or an int")("Error: The dragged object is not in the same stage as the collection. Ensure that objects belong to the same stage before adding them")
```

Comment:
Not sure if there's a better way than ValueError in this case, trying to avoid using Maya or UFE specific methods...

Reply:
In other points of the code there's a 
`raise RuntimeError("Needs to be implemented in derived class")`
So I think the ValueError should be fine here